src/DSUtil/Profile.cpp

```diff
@@ -0,0 +1,987 @@
+	}
+
+	return lResult;
+}
+
+void CProfile::InitIni()
+{
+	std::lock_guard<std::recursive_mutex> lock(m_Mutex);
+
+	if (m_hAppRegKey) {
+		return;
+	}
+
+	// Don't reread mpc-hc.ini if the cache needs to be flushed or it was accessed recently
+	const DWORD tick = GetTickCount();
+	if (m_bIniFirstInit && (m_bIniNeedFlush || tick - m_dwIniLastAccessTick < 100)) {
+		m_dwIniLastAccessTick = tick;
+		return;
+	}
+
+	m_bIniFirstInit = true;
+	m_dwIniLastAccessTick = tick;
+
+	if (!::PathFileExistsW(m_IniPath)) {
+		return;
+	}
+
+	FILE* fp;
+	int fpStatus;
+	do { // Open mpc-hc.ini in UNICODE mode, retry if it is already being used by another process
+		fp = _wfsopen(m_IniPath, L"r, ccs=UNICODE", _SH_SECURE);
+		if (fp || (GetLastError() != ERROR_SHARING_VIOLATION)) {
+			break;
+		}
+		Sleep(100);
+	} while (true);
+	if (!fp) {
+		ASSERT(0);
+		return;
+	}
+	if (_ftell_nolock(fp) == 0L) {
+		// No BOM was consumed, assume mpc-hc.ini is ANSI encoded
+		fpStatus = fclose(fp);
+		ASSERT(fpStatus == 0);
+		do { // Reopen mpc-hc.ini in ANSI mode, retry if it is already being used by another process
+			fp = _wfsopen(m_IniPath, L"r", _SH_SECURE);
+			if (fp || (GetLastError() != ERROR_SHARING_VIOLATION)) {
+				break;
+			}
+			Sleep(100);
+		} while (true);
+		if (!fp) {
+			ASSERT(0);
+			return;
+		}
+	}
+
+	CStdioFile file(fp);
+
+	ASSERT(!m_bIniNeedFlush);
+	m_ProfileMap.clear();
+
+	CStringW line, section, var, val;
+	while (file.ReadString(line)) {
+		// Parse mpc-hc.ini file, this parser:
+		//  - doesn't trim whitespaces
+		//  - doesn't remove quotation marks
+		//  - omits keys with empty names
+		//  - omits unnamed sections
+		int pos = 0;
+		if (line[0] == '[') {
+			pos = line.Find(']');
+			if (pos == -1) {
+				continue;
+			}
+			section = line.Mid(1, pos - 1);
+		} else if (line[0] != ';') {
+			pos = line.Find('=');
+			if (pos == -1) {
+				continue;
+			}
+			var = line.Mid(0, pos);
+			val = line.Mid(pos + 1);
+			if (!section.IsEmpty() && !var.IsEmpty()) {
+				m_ProfileMap[section][var] = val;
+			}
+		}
+	}
+	fpStatus = fclose(fp);
+	ASSERT(fpStatus == 0);
+
+	m_dwIniLastAccessTick = GetTickCount(); // update the last access tick because reading the file can take a long time
+}
+
+bool CProfile::StoreSettingsToRegistry()
+{
+	if (m_hAppRegKey) {
+		DLog(L"StoreSettingsToRegistry(): The settings are already stored in the registry.");
+		return true;
+	}
+
+	InitIni();
+
+	if (::PathFileExistsW(m_IniPath) && _wremove(m_IniPath) != 0) {
+		DLog(L"StoreSettingsToRegistry(): WARNING! The INI file can not be deleted. Perhaps this is protection against changes in settings. Transfer of settings is canceled.");
+		return false;
+	}
+
+	OpenRegistryKey();
+
+	if (!m_hAppRegKey) {
+		InitIni();
+		if (_wremove(m_IniPath) == 0) {
+			DWORD dwDisposition = 0;
+			LONG lResult = RegCreateKeyExW(HKEY_CURRENT_USER, L"Software\\MPC-HC_TEST", 0, nullptr, 0, KEY_READ, nullptr, &m_hAppRegKey, &dwDisposition);
```

Comment:
Why?

Reply:
I can't understand it.  It was MPC-BE_TEST before.  It reads from MPC-BE, but creates MPC-BE_TEST.  Looking at svn log shows that line never has changed since 2018.  I guess it would be ok to remove the _TEST.